mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb

```diff
@@ -585,6 +668,100 @@ func resourceDataflowFlexTemplateJobDelete(d *schema.ResourceData, meta interfac
 }
 
 func resourceDataflowFlexJobTypeCustomizeDiff(_ context.Context, d *schema.ResourceDiff, meta interface{}) error {
+
+	err := dataflowFlexJobTypeParameterOverride("autoscaling_algorithm", "autoscalingAlgorithm", d)
+	if err != nil {
+		return err
+	}
+
+	if p, ok := d.GetOk("parameters.numWorkers"); ok {
```

Comment:
Can you create a test that successfully performs an update? I think we'll hit the error case on any `num_workers` change, because O+C fields will return a value to d.GetOk. I may be missing something, though.

Reply:
I had tested these the most locally since the integers had the funkiest interaction with everything due to Terraform treating a null integer as 0 -- will move to include them more specifically in the update tests

Unless I'm misunderstanding the question --this is why the following line inside checks for `if d.HasChange("num_workers") `, rather than `GetOk("num_workers")`, the O+C value will not be flagged as having a change when customizediff runs unless it is also in the user's configuration (with a value changed from its output read value). If the parameter has been removed from the parameters{} block entirely, then this check to perform a value override is skipped and the rest of the resource update will using `num_workers` as exactly as configured within the user's .tf file (or by persisting the existing computed value if it is never configured).

Reply:
Thanks!